app/scenes/Settings/components/WebhookSubscriptionNew.tsx

```diff
@@ -0,0 +1,292 @@
+import { isEqual } from "lodash";
+import * as React from "react";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { useTranslation, Trans } from "react-i18next";
+import styled from "styled-components";
+import Button from "~/components/Button";
+import { ReactHookWrappedInput } from "~/components/Input";
+import Text from "~/components/Text";
+import useMobile from "~/hooks/useMobile";
+import useStores from "~/hooks/useStores";
+import useToasts from "~/hooks/useToasts";
+
+type Props = {
+  onSubmit: () => void;
+};
+
+interface FormData {
+  name: string;
+  url: string;
+  events: string[];
+}
+
+const EventCheckboxLabel = styled.label`
+  display: flex;
+  align-items: center;
+
+  padding: 0.5rem 0;
+
+  &.group-label {
+    font-weight: bold;
+    font-size: 1.2em;
+  }
+
+  &.all-label {
+    font-weight: bold;
+    font-size: 2em;
+    padding-bottom: 1em;
+  }
+`;
+
+const EventCheckboxText = styled.span`
+  margin-right: 0.5rem;
+`;
+
+const WEBHOOK_EVENTS = {
```

Comment:
This is just a giant structure to hold all the possible events on the Frontend

Thought it would be interesting to try and generate this from the Events on the server side but wasn't sure if mixing across the top-level dirs was a good pattern (or would even work). So I took the easier way and just manually added all the events to the frontend

This was created by looking at the `Events` type here: https://github.com/outline/outline/pull/3607/files#diff-9aa02e81e82dfd51bd95a1dd4046be4847afefab02af12edde04b215e840f2acL231